### ISSUE TITLE
Full job details with retries

### ISSUE THREAD

**Issue by KristofMorva** [laravel/horizon#143]
TL;DR: it would be very useful to see the retried jobs, and all the job details (even if it's just raw JSON or whatever) in the Recent Jobs panel.

A little story for the use case: I have spent the last two days debugging why is a payment charged two times. Horizon listed only one job for `PayLesson` (taken 21 seconds), but duplicates for related invoicing/notifying/etc, so I started to check the code step-by-step, how is it possible to reach the payment part two times. After suffering for some time, I have checked the supervisor log, and I have seen the duplicate processing of `PayLesson`:
```
[2017-09-14 20:00:05] Processing:     App\Jobs\PayLesson
[2017-09-14 20:00:35] Processing:     App\Jobs\PayLesson
[2017-09-14 20:00:47] Processed:      App\Jobs\PayLesson
... bunch of shit ...
[2017-09-14 20:00:56] Processed:      App\Jobs\PayLesson
```
I could have been smarter, because I didn't realize exactly 30 seconds passed between the two (which is very likely the default timeout), so after some hours of frustration I have decided to dump out the whole job to the `WorkCommand.php` like this: `$job->resolveName() . ' - ' . $job->getRawBody()`. The result was enlightening, as I have seen it has been retried, then I checked "why was it immediately retried?", and I have seen the 30 seconds:
```
[2017-09-14 20:00:05] Processing:     App\Jobs\PayLesson - {"displayName":"App\\Jobs\\PayLesson","job":"Illuminate\\Queue\\CallQueuedHandler@call","maxTries":null,"timeout":null,"timeoutAt":null,"data":{"commandName":"App\\Jobs\\PayLesson","command":"O:18:\"App\\Jobs\\PayLesson\":6:{s:9:\"\u0000*\u0000lesson\";O:45:\"Illuminate\\Contracts\\Database\\ModelIdentifier\":3:{s:5:\"class\";s:17:\"App\\Models\\Lesson\";s:2:\"id\";i:9;s:10:\"connection\";s:5:\"mysql\";}s:6:\"\u0000*\u0000job\";N;s:10:\"connection\";N;s:5:\"queue\";N;s:5:\"delay\";N;s:7:\"chained\";a:0:{}}"},"id":"25","attempts":0,"type":"job","tags":["App\\Models\\Lesson:9"],"pushedAt":1505419202}
[2017-09-14 20:00:35] Processing:     App\Jobs\PayLesson - {"type":"job","timeout":null,"tags":["App\\Models\\Lesson:9"],"id":"25","data":{"command":"O:18:\"App\\Jobs\\PayLesson\":6:{s:9:\"\u0000*\u0000lesson\";O:45:\"Illuminate\\Contracts\\Database\\ModelIdentifier\":3:{s:5:\"class\";s:17:\"App\\Models\\Lesson\";s:2:\"id\";i:9;s:10:\"connection\";s:5:\"mysql\";}s:6:\"\u0000*\u0000job\";N;s:10:\"connection\";N;s:5:\"queue\";N;s:5:\"delay\";N;s:7:\"chained\";a:0:{}}","commandName":"App\\Jobs\\PayLesson"},"displayName":"App\\Jobs\\PayLesson","timeoutAt":null,"pushedAt":1505419202,"maxTries":null,"job":"Illuminate\\Queue\\CallQueuedHandler@call","attempts":1}
[2017-09-14 20:00:47] Processed:      App\Jobs\PayLesson - {"displayName":"App\\Jobs\\PayLesson","job":"Illuminate\\Queue\\CallQueuedHandler@call","maxTries":null,"timeout":null,"timeoutAt":null,"data":{"commandName":"App\\Jobs\\PayLesson","command":"O:18:\"App\\Jobs\\PayLesson\":6:{s:9:\"\u0000*\u0000lesson\";O:45:\"Illuminate\\Contracts\\Database\\ModelIdentifier\":3:{s:5:\"class\";s:17:\"App\\Models\\Lesson\";s:2:\"id\";i:9;s:10:\"connection\";s:5:\"mysql\";}s:6:\"\u0000*\u0000job\";N;s:10:\"connection\";N;s:5:\"queue\";N;s:5:\"delay\";N;s:7:\"chained\";a:0:{}}"},"id":"25","attempts":0,"type":"job","tags":["App\\Models\\Lesson:9"],"pushedAt":1505419202}
... bunch of shit ...
[2017-09-14 20:00:56] Processed:      App\Jobs\PayLesson - {"type":"job","timeout":null,"tags":["App\\Models\\Lesson:9"],"id":"25","data":{"command":"O:18:\"App\\Jobs\\PayLesson\":6:{s:9:\"\u0000*\u0000lesson\";O:45:\"Illuminate\\Contracts\\Database\\ModelIdentifier\":3:{s:5:\"class\";s:17:\"App\\Models\\Lesson\";s:2:\"id\";i:9;s:10:\"connection\";s:5:\"mysql\";}s:6:\"\u0000*\u0000job\";N;s:10:\"connection\";N;s:5:\"queue\";N;s:5:\"delay\";N;s:7:\"chained\";a:0:{}}","commandName":"App\\Jobs\\PayLesson"},"displayName":"App\\Jobs\\PayLesson","timeoutAt":null,"pushedAt":1505419202,"maxTries":null,"job":"Illuminate\\Queue\\CallQueuedHandler@call","attempts":1}
```
Being able to see it in Horizon (in my opinion) would make it much more useful for debugging.

**Comment by fgilio** [comment]
This would be awesome, had a similar situation some days ago. It'd be really useful to have more visibility onto the retries and the jobs data.

**Comment by Muffinman** [comment]
I have a use case where some jobs are taking a lot longer to process than others, and this feature request would really improve my qualify of life!

In my case I'd really like to inspect the parameters passed into the job, and it would be really handy to do so directly in Horizon without adding custom logging or having to break out a redis GUI.

Apart from showing the name of the job and the tags, I feel the current interface is missing quite a bit of info which is readily available in redis.

![image](https://user-images.githubusercontent.com/1319568/66110521-dadd6480-e5be-11e9-90d3-b5ec76f6be9f.png)

I feel clicking on the job here could expand a div contianing extra JSON data, link to a new page, or perhaps even use a modal?

**Comment by brunogaspar** [comment]
Basically the same behaviour as if you clicked on a failed job @Muffinman ?

**Comment by Muffinman** [comment]
Yes that would be great :)

Although even that does not show the arguments passed to the job, but in my case they get assigned to local properties so I can inspect them from there.

**Comment by brunogaspar** [comment]
ah like that.

Never had the need to inspect those to be honest, as all i need i associate to properties on the class but should be possible, maybe?

@driesvints Wanna shed some light here and determine if this is something doable to be added?

**Comment by driesvints** [comment]
Yeah I think so. Feel free to send in a PR.

**Comment by Muffinman** [comment]
Going to send a PR for this, do you want it based against `3.0` or `master`?

**Comment by driesvints** [comment]
Closing this as the pr was rejected by Taylor. Sorry.
